applications/nrf5340_audio/src/modules/button_handler.c

```diff
@@ -137,21 +161,16 @@ static void button_isr(const struct device *port, struct gpio_callback *cb, uint
 	LOG_DBG("Pushed button idx: %d pin: %d name: %s", btn_idx, btn_pin,
 		btn_cfg[btn_idx].btn_name);
 
-	event.button_activity.button_pin = btn_pin;
-	event.button_activity.button_action = BUTTON_PRESS;
-	event.event_source = EVT_SRC_BUTTON;
-
-	/* To avoid filling up the event queue with button presses,
-	 * we only allow button events if all other events have been processed
-	 */
-	if (ctrl_events_queue_empty()) {
-		ret = ctrl_events_put(&event);
-		ERR_CHK(ret);
-		debounce_is_ongoing = true;
-		k_timer_start(&button_debounce_timer, K_MSEC(CONFIG_BUTTON_DEBOUNCE_MS), K_NO_WAIT);
-	} else {
-		LOG_WRN("Event queue is not empty, try again later");
+	msg.button_pin = btn_pin;
+	msg.button_action = BUTTON_PRESS;
+
+	ret = k_msgq_put(&button_queue, (void *)&msg, K_NO_WAIT);
+	if (ret == -EAGAIN) {
+		LOG_WRN("Btn msg queue full");
 	}
+
+	debounce_is_ongoing = true;
```

Comment:
So now there are two debouncing functionalities?
This + K_MSGQ_DEFINE(button_queue, sizeof(struct button_msg), 1, 4); on line 30?

Reply:
Kind of, but the most important is the boolean with the timer.